test/functional/ex_cmds/shell_filter_spec.lua

```diff
@@ -0,0 +1,35 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local insert = helpers.insert
+local expect = helpers.expect
+local feed = helpers.feed
+
+before_each(clear)
+
+describe('shell filtering', function()
```

Comment:
don't want a separate test file for this. why move it?

Reply:
I just think this is unrelated to `system()`

Reply:
ok, we could rename  system_spec.lua to shell_spec and maybe add a describe('shell') group. I have some local changes, so please avoid pushing to the PR.